orchprovider/k8s/v1/k8s.go

```diff
@@ -117,23 +151,54 @@ func k8sOrchUtil(k *k8sOrchestrator, volProfile volProfile.VolumeProvisionerProf
 	return k.k8sUtlGtr.GetK8sUtil(volProfile)
 }
 
-// TODO
-//    Will it be good if this accepts VolumeProvisionerProfile and updates the
-// k8sOrchestrator instance's properties ?
-//
-// StorageOps provides storage operations instance that deals with all storage
-// related functionality by aligning with Kubernetes as the orchestration provider.
-//
-// NOTE:
-//    This is an implementation of the orchprovider.OrchestratorInterface interface.
-//
-// NOTE:
-//    This is invoked on a per request basis. In other words, every request will
-// invoke StorageOps to invoke storage specific operations thereafter.
+// StorageOps provides volume operations instance.
 func (k *k8sOrchestrator) StorageOps() (orchprovider.StorageOps, bool) {
 	return k, true
 }
 
+// PolicyOps provides a policy operations instance.
+// In addition, it is used for various initializations & validations
+func (k *k8sOrchestrator) PolicyOps(vol *v1.Volume) (orchprovider.PolicyOps, bool, error) {
+	err := k.setVolume(vol)
+	if err != nil {
+		return nil, true, err
+	}
+
+	err = k.setK8sUtil(&k8sUtil{
+		volume: vol,
+	})
+	if err != nil {
+		return nil, true, err
+	}
+
+	return k, true, nil
+}
+
+// FetchPolicies will fetch volume policies based on the volume
+func (k *k8sOrchestrator) FetchPolicies() (map[string]string, error) {
+	kc, supported, err := k.k8sUtil.K8sClientV2()
+	if err != nil {
+		return nil, err
+	}
+
+	if !supported {
+		return nil, fmt.Errorf("K8s client is not supported")
+	}
+
+	// fetch k8s Deployment operator
+	scOps, err := kc.StorageClassOps()
+	if err != nil {
+		return nil, err
+	}
+
+	sc, err := scOps.Get(k.volume.Labels.K8sStorageClass, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return sc.Parameters, nil
+}
+
 // AddStorage will add persistent volume running as containers. In OpenEBS
 // terms AddStorage will add a VSM.
```

Comment:
will as a Volume ?

Reply:
@prateekpandey14 Did not understand the query

Reply:
i.e. VSM --> Volume